src/libraries/System.Net.Ping/src/System/Net/NetworkInformation/Ping.cs

```diff
@@ -318,6 +324,28 @@ public Task<PingReply> SendPingAsync(IPAddress address, int timeout, byte[] buff
             return SendPingAsyncInternal(address, timeout, buffer, options);
         }
 
+        public Task<PingReply> SendPingAsync(IPAddress address, TimeSpan timeout, byte[]? buffer = null,
+            PingOptions? options = null, CancellationToken cancellationToken = default)
+        {
+            int milliseconds = ToTimeoutMilliseconds(timeout);
```

Comment:
nit, this and the next test call ToTimeoutMilliseconds and ThrowIfCancellationRequested in inconsistent orders. It makes no difference, but it is a speed bump when reading.

I would just put the call to ToTimeoutMilliseconds directly in the method call, eliminate the variable.